src/box/box.cc

```diff
@@ -2832,10 +2835,19 @@ box_wait_quorum(uint32_t lead_id, int64_t target_lsn, int quorum,
 		trigger_create(&t.base, box_quorum_on_ack_f, NULL, NULL);
 		trigger_add(&replicaset.on_ack, &t.base);
 		double deadline = ev_monotonic_now(loop()) + timeout;
-		do {
-			if (fiber_yield_deadline(deadline))
+		while (!fiber_is_cancelled() && t.ack_count < t.quorum) {
```

Comment:
I propose to add a warning in `box_wait_quorum`, when it's started with quorum more than the number of `registered_count`. Otherwise, if it happens on production it's non-obvious, why the node is still in RO, if it's a leader already

Reply:
There's no need to warn user on evert wakeup) It's enough to say it once, when the function is started, IMHO